components/batch-system/benches/batch-system.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+#![feature(test)]
+
+extern crate test;
+
+use batch_system::test_runner::*;
+use batch_system::*;
+use criterion::*;
+use std::sync::atomic::*;
+use std::sync::Arc;
+
+fn end_hook(tx: &std::sync::mpsc::Sender<()>) -> Message {
+    let tx = tx.clone();
+    Message::Callback(Box::new(move |_| {
+        tx.send(()).unwrap();
+    }))
+}
+
+fn bench_spawn_many(c: &mut Criterion) {
+    let (control_tx, control_fsm) = Runner::new(100000);
+    let (router, mut system) = batch_system::create_system(2, 2, control_tx, control_fsm);
+    system.spawn("test".to_owned(), Builder::new());
+    const ID_LIMIT: u64 = 32;
+    const MESSAGE_LIMIT: usize = 256;
+    for id in 0..ID_LIMIT {
+        let (normal_tx, normal_fsm) = Runner::new(100000);
+        let normal_box = BasicMailbox::new(normal_tx, normal_fsm);
+        router.register(id, normal_box);
+    }
+
+    let (tx, rx) = std::sync::mpsc::channel();
+    c.bench_function("spawn_many", |b| {
+        b.iter(|| {
+            for id in 0..ID_LIMIT {
+                for i in 0..MESSAGE_LIMIT {
+                    router.send(id, Message::Loop(i)).unwrap();
+                }
+                router.send(id, end_hook(&tx)).unwrap();
+            }
+            for _ in 0..ID_LIMIT {
+                rx.recv().unwrap();
+            }
+        })
+    });
+    system.shutdown();
+}
+
+fn bench_imbalance(c: &mut Criterion) {
```

Comment:
Could these top level functions get some docs to describe what is being tested/benchmarked?